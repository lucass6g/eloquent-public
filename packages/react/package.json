{
  "name": "@eloquent/react",
  "version": "1.0.0",
  "description": "",
  "main": "./dist/index.js",
  "files": [
    "dist"
  ],
  "types": "./dist/index.d.ts",
  "exports": {
    ".": {
      "import": "./dist/index.mjs",
      "require": "./dist/index.js",
      "types": "./dist/index.d.ts"
    }
  },
  "scripts": {
    "build": "tsup",
    "clean": "rm -rf dist && rm -rf node_modules"
  },
  "keywords": [],
  "author": "",
  "license": "MIT",
  "devDependencies": {
    "tsup": "^8.1.0",
    "@types/react": "^18.3.3",
    "@types/react-dom": "^18.3.0",
    "react": "^18.3.1",
    "typescript": "^5.4.5"
  }
}
