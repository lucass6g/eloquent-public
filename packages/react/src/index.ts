export {Switch, Case, Default} from './components/conditional-switch'