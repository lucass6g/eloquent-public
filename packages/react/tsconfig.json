{
  "extends": "../../packages/config/base.tsconfig.json",
  "include": ["src"],
  "exclude": ["dist", "build", "node_modules"],
  "compilerOptions": {
    "forceConsistentCasingInFileNames": false,
    "rootDir": "src",
    "baseUrl": ".",
    "paths": {
      "@/*": ["./src/*"]
    }
  }
}
