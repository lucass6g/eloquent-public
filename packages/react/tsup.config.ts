import { defineConfig } from 'tsup';

export default defineConfig({
	name: "@eloquent/react",
	entry: ["src/index.ts"],
	outDir: "dist",
	target: "es2022",
	format: ["cjs", "esm"],
	splitting: false,
	sourcemap: true,
	clean: true,
	dts: true,
	external: ["react", 'react-dom', 'react/jsx-runtime'],
	cjsInterop: true,
	treeshake: true,
})