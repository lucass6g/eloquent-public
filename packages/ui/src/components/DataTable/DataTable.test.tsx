import { it, describe, expect } from "vitest";
import { render } from "@testing-library/react";
import { Pagination, Table } from "../";
import { DataTable } from ".";

describe("Data Table", () => {
  it("should match the snapshot", () => {
    const component = render(
      <DataTable.Root>
        <Table.Root>
          <Table.Caption>A test table</Table.Caption>
          <Table.Header>
            <Table.Row>
              <Table.Head>Header 1</Table.Head>
              <Table.Head>Header 2</Table.Head>
            </Table.Row>
          </Table.Header>
          <Table.Body>
            <Table.Row>
              <Table.Cell>Cell 1</Table.Cell>
              <Table.Cell>Cell 2</Table.Cell>
            </Table.Row>
          </Table.Body>
          <Table.Footer>
            <Table.Row>
              <Table.Cell>Total</Table.Cell>
              <Table.Cell>$2,500.00</Table.Cell>
            </Table.Row>
          </Table.Footer>
        </Table.Root>
        <Pagination.Root className="flex items-center gap-4">
          <Pagination.Label current={1} total={10} />
          <Pagination.Content>
            <Pagination.Item>
              <Pagination.FirstPage disabled={true} />
            </Pagination.Item>
            <Pagination.Item>
              <Pagination.Previous onClick={() => console.log("previous")} />
            </Pagination.Item>
            <Pagination.Item>
              <Pagination.Next onClick={() => console.log("next")} />
            </Pagination.Item>
            <Pagination.Item>
              <Pagination.LastPage onClick={() => console.log("last")} />
            </Pagination.Item>
          </Pagination.Content>
        </Pagination.Root>
      </DataTable.Root>
    );

    expect(component.baseElement).toMatchSnapshot();
  });
});
