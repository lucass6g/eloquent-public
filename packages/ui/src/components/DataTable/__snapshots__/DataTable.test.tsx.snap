// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Data Table > should match the snapshot 1`] = `
<body>
  <div>
    <div
      class="flex flex-col items-end"
    >
      <div
        class="relative w-full overflow-auto"
      >
        <table
          class="w-full caption-bottom text-xs"
        >
          <caption
            class="mt-4 text-xs text-neutral-800 font-normal text-center"
          >
            A test table
          </caption>
          <thead
            class="bg-neutral-200 h-12 overflow-hidden"
          >
            <tr
              class="border-t transition-colors first-of-type:border-0"
            >
              <th
                class="first-of-type:rounded-l-[10px] last-of-type:rounded-r-[10px] px-4 text-left align-middle text-neutral-500 font-medium text-sm"
              >
                Header 1
              </th>
              <th
                class="first-of-type:rounded-l-[10px] last-of-type:rounded-r-[10px] px-4 text-left align-middle text-neutral-500 font-medium text-sm"
              >
                Header 2
              </th>
            </tr>
          </thead>
          <tbody
            class="[&_tr]:h-16"
          >
            <tr
              class="border-t transition-colors first-of-type:border-0"
            >
              <td
                class="px-4 align-middle text-neutral-800 font-normal text-sm"
              >
                Cell 1
              </td>
              <td
                class="px-4 align-middle text-neutral-800 font-normal text-sm"
              >
                Cell 2
              </td>
            </tr>
          </tbody>
          <tfoot
            class="border-t font-medium [&>tr]:last:border-b-0"
          >
            <tr
              class="border-t transition-colors first-of-type:border-0"
            >
              <td
                class="px-4 align-middle text-neutral-800 font-normal text-sm"
              >
                Total
              </td>
              <td
                class="px-4 align-middle text-neutral-800 font-normal text-sm"
              >
                $2,500.00
              </td>
            </tr>
          </tfoot>
        </table>
      </div>
      <nav
        aria-label="pagination"
        class="flex items-center gap-4"
        role="navigation"
      >
        <span
          class="text-accent-foreground font-normal text-sm"
        >
          Página 
          1
           de 
          10
        </span>
        <ul
          class="flex flex-row items-center gap-1"
        >
          <li>
            <button
              aria-label="Primeira página"
              class="inline-flex items-center justify-center whitespace-nowrap rounded-lg transition-colors focus-visible:outline-none focus-visible:ring-1 focus-visible:ring-ring focus-visible:ring-offset-1 disabled:cursor-not-allowed text-sm antialiased leading-4 p-0 border border-border h-10 w-10 first-child:stroke-primary-foreground gap-1 bg-neutral-100 disabled:bg-neutral-100 shadow-none disabled:text-neutral-400 text-neutral-800 hover:bg-neutral-200 font-normal data-[active=true]:bg-neutral-200 data-[active=true]:font-semibold"
              data-active="false"
              disabled=""
            >
              <svg
                class="lucide lucide-chevrons-left h-4 w-4 stroke-[1.5px]"
                fill="none"
                height="24"
                stroke="currentColor"
                stroke-linecap="round"
                stroke-linejoin="round"
                stroke-width="2"
                viewBox="0 0 24 24"
                width="24"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="m11 17-5-5 5-5"
                />
                <path
                  d="m18 17-5-5 5-5"
                />
              </svg>
            </button>
          </li>
          <li>
            <button
              aria-label="Página anterior"
              class="inline-flex items-center justify-center whitespace-nowrap rounded-lg transition-colors focus-visible:outline-none focus-visible:ring-1 focus-visible:ring-ring focus-visible:ring-offset-1 disabled:cursor-not-allowed text-sm antialiased leading-4 p-0 border border-border h-10 w-10 first-child:stroke-primary-foreground gap-1 bg-neutral-100 disabled:bg-neutral-100 shadow-none disabled:text-neutral-400 text-neutral-800 hover:bg-neutral-200 font-normal data-[active=true]:bg-neutral-200 data-[active=true]:font-semibold"
              data-active="false"
            >
              <svg
                class="lucide lucide-chevron-left h-4 w-4 stroke-[1.5px]"
                fill="none"
                height="24"
                stroke="currentColor"
                stroke-linecap="round"
                stroke-linejoin="round"
                stroke-width="2"
                viewBox="0 0 24 24"
                width="24"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="m15 18-6-6 6-6"
                />
              </svg>
            </button>
          </li>
          <li>
            <button
              aria-label="Próxima página"
              class="inline-flex items-center justify-center whitespace-nowrap rounded-lg transition-colors focus-visible:outline-none focus-visible:ring-1 focus-visible:ring-ring focus-visible:ring-offset-1 disabled:cursor-not-allowed text-sm antialiased leading-4 p-0 border border-border h-10 w-10 first-child:stroke-primary-foreground gap-1 bg-neutral-100 disabled:bg-neutral-100 shadow-none disabled:text-neutral-400 text-neutral-800 hover:bg-neutral-200 font-normal data-[active=true]:bg-neutral-200 data-[active=true]:font-semibold"
              data-active="false"
            >
              <svg
                class="lucide lucide-chevron-right h-4 w-4 stroke-[1.5px]"
                fill="none"
                height="24"
                stroke="currentColor"
                stroke-linecap="round"
                stroke-linejoin="round"
                stroke-width="2"
                viewBox="0 0 24 24"
                width="24"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="m9 18 6-6-6-6"
                />
              </svg>
            </button>
          </li>
          <li>
            <button
              aria-label="Última página"
              class="inline-flex items-center justify-center whitespace-nowrap rounded-lg transition-colors focus-visible:outline-none focus-visible:ring-1 focus-visible:ring-ring focus-visible:ring-offset-1 disabled:cursor-not-allowed text-sm antialiased leading-4 p-0 border border-border h-10 w-10 first-child:stroke-primary-foreground gap-1 bg-neutral-100 disabled:bg-neutral-100 shadow-none disabled:text-neutral-400 text-neutral-800 hover:bg-neutral-200 font-normal data-[active=true]:bg-neutral-200 data-[active=true]:font-semibold"
              data-active="false"
            >
              <svg
                class="lucide lucide-chevrons-right h-4 w-4 stroke-[1.5px]"
                fill="none"
                height="24"
                stroke="currentColor"
                stroke-linecap="round"
                stroke-linejoin="round"
                stroke-width="2"
                viewBox="0 0 24 24"
                width="24"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="m6 17 5-5-5-5"
                />
                <path
                  d="m13 17 5-5-5-5"
                />
              </svg>
            </button>
          </li>
        </ul>
      </nav>
    </div>
  </div>
</body>
`;
